WordPress/Classes/ViewRelated/Stats/Charts/Charts+AxisFormatters.swift

```diff
@@ -65,13 +65,44 @@ class VerticalAxisFormatter: IAxisValueFormatter {
     // MARK: IAxisValueFormatter
 
     func stringForValue(_ value: Double, axis: AxisBase?) -> String {
-        let number = NSNumber(value: value/1000)
-        let value = formatter.string(from: number) ?? ""
+        let threshold = Double(1000)
 
-        // This is, admittedly NOT locale-sensitive formatting approach.
-        // It is slated to be improved via #11143.
-        let formattedValue = "\(value)k"
+        let formattedValue: String
+        if value > threshold {
+            let numericValue = NSNumber(value: value/threshold)
+            let rawFormattedValue = formatter.string(from: numericValue) ?? "\(numericValue)"
+
+            // This is, admittedly NOT locale-sensitive formatting approach. It will be improved via #11143.
+            formattedValue = "\(rawFormattedValue)k"
+        } else {
+            let numericValue = NSNumber(value: value)
+            formattedValue = formatter.string(from: numericValue) ?? "\(value)"
+        }
 
         return formattedValue
+
+        // Take 1
+//        let number = NSNumber(value: value/1000)
+//        let value = formatter.string(from: number) ?? ""
+//
+//        // This is, admittedly NOT locale-sensitive formatting approach.
+//        // It is slated to be improved via #11143.
+//        let formattedValue = "\(value)k"
+//
+//        return formattedValue
+
+        // Take 2
+//        let formattedValue: String
+//        let threshold = Double(1000)
+//        if value > threshold {
+//            // This is, admittedly NOT locale-sensitive formatting approach. It will be improved via #11143.
+//            let scaledValue = NSNumber(value: value/threshold)
+//            let rawFormattedValue = formatter.string(from: scaledValue) ?? ""
+//            formattedValue = "\(rawFormattedValue)k"
+//        } else {
+//            formattedValue = formatter.string(from: value) ?? "\(value)"
+//        }
+//
+//        return formattedValue
```

Comment:
I'm assuming this block of commented out code can be removed?

Reply:
Absolutely - thanks for catching that!